x/protorev/keeper/epoch_hook.go

```diff
@@ -109,28 +104,26 @@ func (k Keeper) GetHighestLiquidityPools(ctx sdk.Context) (map[string]LiquidityP
 				Liquidity: tokenA.Amount.Mul(tokenB.Amount),
 			}
 
-			// Check if there is a match with osmo
-			if otherDenom, match := types.CheckOsmoAtomDenomMatch(tokenA.Denom, tokenB.Denom, types.OsmosisDenomination); match {
-				k.updateHighestLiquidityPool(otherDenom, osmoPools, newPool)
+			// Update happens both ways to ensure the pools that contain multiple base denoms are properly updated
+			if pools, ok := baseDenomPools[tokenA.Denom]; ok {
+				k.updateHighestLiquidityPool(tokenB.Denom, pools, newPool)
 			}
-
-			// Check if there is a match with atom
-			if otherDenom, match := types.CheckOsmoAtomDenomMatch(tokenA.Denom, tokenB.Denom, types.AtomDenomination); match {
-				k.updateHighestLiquidityPool(otherDenom, atomPools, newPool)
+			if pools, ok := baseDenomPools[tokenB.Denom]; ok {
```

Comment:
Same thing here (Prob good to change the pools to a diff var name since the main loop ranges through pools)